src/components/SwapForm/index.tsx

```diff
@@ -0,0 +1,250 @@
+  const [recipient, setRecipient] = useState<string | null>(null)
+  const [isSaveGas, setSaveGas] = useState(false)
+
+  const parsedAmount = useParsedAmount(currencyIn, typedValue)
+  const { wrapType, inputError: wrapInputError, execute: onWrap } = useWrapCallback(currencyIn, currencyOut, typedValue)
+  const isWrapOrUnwrap = wrapType !== WrapType.NOT_APPLICABLE
+
+  const { fetcher: getRoute, result } = useGetRoute({
+    currencyIn,
+    currencyOut,
+    feeConfig,
+    isSaveGas,
+    parsedAmount,
+  })
+
+  const { data: getRouteRawResponse, isFetching: isGettingRoute, error: getRouteError } = result
+  const getRouteResponse = useMemo(() => {
+    if (!getRouteRawResponse?.data || getRouteError || !currencyIn || !currencyOut) {
+      return undefined
+    }
+
+    return parseGetRouteResponse(getRouteRawResponse.data, currencyIn, currencyOut)
+  }, [currencyIn, currencyOut, getRouteError, getRouteRawResponse])
+
+  const routeSummary = getRouteResponse?.routeSummary
+
+  const buildRoute = useBuildRoute({
+    recipient: isAdvancedMode && recipient ? recipient : '',
+    routeSummary: getRouteRawResponse?.data?.routeSummary || undefined,
+    slippage,
+    transactionTimeout,
+    skipSimulateTx: isAdvancedMode,
+  })
+
+  const swapInputError = useGetInputError({
+    currencyIn,
+    currencyOut,
+    typedValue,
+    recipient,
+    balanceIn,
+    parsedAmountFromTypedValue: parsedAmount,
+  })
+
+  const handleChangeCurrencyIn = (c: Currency) => {
+    setRouteSummary(undefined)
+    onChangeCurrencyIn(c)
+  }
+
+  const handleChangeCurrencyOut = (c: Currency) => {
+    setRouteSummary(undefined)
+    onChangeCurrencyOut(c)
+  }
+
+  const isSolanaUnwrap = isSolana && wrapType === WrapType.UNWRAP
+  useEffect(() => {
+    // reset value for unwrapping WSOL
+    // because on Solana, unwrap WSOL is closing WSOL account,
+    // which mean it will unwrap all WSOL at once and we can't unwrap partial amount of WSOL
+    if (isSolanaUnwrap) setTypedValue(balanceIn?.toExact() ?? '')
+  }, [balanceIn, isSolanaUnwrap])
+
+  useEffect(() => {
+    setRouteSummary(routeSummary)
+  }, [routeSummary, setRouteSummary])
+
+  return (
+    <SwapFormContextProvider
+      feeConfig={feeConfig}
+      slippage={slippage}
+      routeSummary={routeSummary}
+      typedValue={typedValue}
+      isSaveGas={isSaveGas}
+      recipient={recipient}
+    >
+      <Flex sx={{ flexDirection: 'column', gap: '16px' }}>
+        <Wrapper id={TutorialIds.SWAP_FORM_CONTENT}>
+          <Flex flexDirection="column" sx={{ gap: '0.75rem' }}>
+            <InputCurrencyPanel
+              wrapType={wrapType}
+              typedValue={typedValue}
+              setTypedValue={setTypedValue}
+              routeSummary={routeSummary}
+              currencyIn={currencyIn}
+              currencyOut={currencyOut}
+              balanceIn={balanceIn}
+              onChangeCurrencyIn={handleChangeCurrencyIn}
+            />
```

Comment:
<img width="571" alt="image" src="https://user-images.githubusercontent.com/11427213/222354473-c0de4f32-8f86-4bb7-abda-bd8e126b1e05.png">

maybe this is duplicated